ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -528,8 +529,15 @@ public static Map<String, Object> buildStreamsProperties(
       final KsqlConfig config,
       final ProcessingLogContext processingLogContext
   ) {
-    final Map<String, Object> newStreamsProperties
-        = new HashMap<>(config.getKsqlStreamConfigProps(applicationId));
+    final Map<String, Object> newStreamsProperties;
+
+    if (Objects.equals(config.getString(KSQL_DEPLOYMENT_TYPE_CONFIG),
+        KsqlConfig.DeploymentType.cloud.toString())) {
+      newStreamsProperties = new HashMap<>(config.getKsqlStreamConfigProps(applicationId, false));
+    } else {
+      newStreamsProperties = new HashMap<>(config.getKsqlStreamConfigProps(applicationId, true));
+    }
```

Comment:
If/else isn't needed here since you can just use the boolean expression directly.

If you're worried about the readability I think it's better just to create a temporary named variable for the boolean.
```suggestion
    boolean isCloudDeployment = KsqlConfig.DeploymentType.cloud.toString().equals(config.getString(KSQL_DEPLOYMENT_TYPE_CONFIG))
    newStreamsProperties = new HashMap<>(config.getKsqlStreamConfigProps(applicationId, !isCloudDeployment));
```